test/peer.browser.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { mockConnection, mockMultiaddrConnection, mockRegistrar, mockStream, mockUpgrader } from '@libp2p/interface-mocks'
+import { createEd25519PeerId } from '@libp2p/peer-id-factory'
+import { multiaddr } from '@multiformats/multiaddr'
+import { expect } from 'aegir/chai'
+import { pair } from 'it-pair'
+import { duplexPair } from 'it-pair/duplex'
+import { pbStream } from 'it-pb-stream'
+import sinon from 'sinon'
+import { connect, handleIncomingStream } from '../src/peer_transport/handler'
+import { Message } from '../src/peer_transport/pb/index.js'
+import { WebRTCDirectTransport } from '../src/peer_transport/transport'
+import { detect } from 'detect-browser'
+
+const browser = detect()
+
+describe('webrtc basic', () => {
+  const runner: any = ((browser != null) && browser.name !== 'firefox') ? it : it.skip
```

Comment:
It does. Firefox is odd since it did not support the `connectionState` instance property. Refer [here](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/connectionState#browser_compatibility). This has been fixed.